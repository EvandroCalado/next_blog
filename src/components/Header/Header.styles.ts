import styled, { css } from 'styled-components';

export const Container = styled.header`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    padding: ${theme.spacings.medium};
    margin-bottom: ${theme.spacings.large};
    text-align: center;

    a {
      display: flex;
      align-items: center;
      justify-content: center;
      gap: ${theme.spacings.medium};
    }
  `}
`;

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 10rem;
    border-radius: 50%;
    border: 2px solid ${theme.colors.secondary};
  `}
`;
