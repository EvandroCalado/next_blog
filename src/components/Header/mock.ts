export default {
  image: 'https://avatars.githubusercontent.com/u/110628201?v=4',
  title: 'Evandro Calado',
};
