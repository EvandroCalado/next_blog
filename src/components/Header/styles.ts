import styled, { css } from 'styled-components';
import { Button } from '../ToggleButton/styles';

export const Header = styled.header`
  ${({ theme }) => css`
    width: 100%;
    background-color: ${theme.colors.primary};
    padding: ${theme.spacings.medium} 0;
    margin-bottom: ${theme.spacings.large};
    transition: all 300ms ease-in-out;
  `}
`;

export const Wrapper = styled.div`
${({ theme }) => css`
    position: relative;
    width: 100%;
    max-width: 120rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.medium};
    display: flex;
    justify-content: space-between;

    ${Button} {
      position: absolute;
      right: ${theme.spacings.medium};
      bottom: 0;
    }
  `}
  
`;
