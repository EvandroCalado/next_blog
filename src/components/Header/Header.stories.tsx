import Header from './Header';
import { Meta, StoryFn } from '@storybook/react';
import { HeaderProps } from './Header';

import mock from './mock';

export default {
  title: 'Header',
  component: Header,
  args: mock,
} as Meta;

export const Template: StoryFn<HeaderProps> = (args) => (
  <div>
    <Header {...args} />
  </div>
);
