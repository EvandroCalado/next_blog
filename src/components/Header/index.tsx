import * as Styled from './styles';
import Search from '../Search';
import Logo from '../Logo';
import ToggleButton from '../ToggleButton';

export type HeaderProps = {
  image: string;
  title: string;
  description: string;
};

const Header = ({ image, title, description }: HeaderProps) => {
  return (
    <Styled.Header>
      <Styled.Wrapper>
        <Logo image={image} title={title} description={description} />
        <Search />
        <ToggleButton />
      </Styled.Wrapper>
    </Styled.Header>
  );
};

export default Header;
