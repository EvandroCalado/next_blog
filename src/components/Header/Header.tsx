import Link from 'next/link';
import * as Styled from './Header.styles';

export type HeaderProps = {
  image: string;
  title: string;
};

const Header = ({ image, title }: HeaderProps) => {
  return (
    <Styled.Container>
      <Link href={'/'}>
        <Styled.Image src={image} alt={title} />
        <Styled.Title>Evandro Calado</Styled.Title>
      </Link>
    </Styled.Container>
  );
};

export default Header;
