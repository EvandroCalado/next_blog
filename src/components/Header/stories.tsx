import Header from '.';
import { Meta, StoryFn } from '@storybook/react';
import { HeaderProps } from '.';

import mock from './mock';

export default {
  title: 'Header',
  component: Header,
  args: mock,
} as Meta;

export const Template: StoryFn<HeaderProps> = (args) => (
  <div>
    <Header {...args} />
  </div>
);
