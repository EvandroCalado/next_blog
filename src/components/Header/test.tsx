import Header from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import mock from './mock';

describe('<Header />', () => {
  it('test_logo_receives_props', () => {
    renderTheme(<Header {...mock} />);

    expect(screen.getByAltText(mock.title)).toBeInTheDocument();
    expect(screen.getByText(mock.title)).toBeInTheDocument();
    expect(screen.getByText(mock.description)).toBeInTheDocument();
  });

  it('should_to_match_snapshot', () => {
    const { container } = renderTheme(<Header {...mock} />);

    expect(container).toMatchSnapshot();
  });
});
