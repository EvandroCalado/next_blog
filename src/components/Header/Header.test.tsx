import Header from './Header';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<Header />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Header {...mock} />);
    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute('href', '/');
    expect(container).toMatchSnapshot();
  });
});
