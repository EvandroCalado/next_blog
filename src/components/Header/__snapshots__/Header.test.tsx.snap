// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > should render 1`] = `
.c0 {
  background-color: #010101;
  color: #FFFFFF;
  font-size: 2.4rem;
  padding: 2.4rem;
  margin-bottom: 3.2rem;
  text-align: center;
}

.c0 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 2.4rem;
}

.c2 {
  font-size: 3.2rem;
}

.c1 {
  width: 10rem;
  border-radius: 50%;
  border: 2px solid #005b96;
}

<div>
  <header
    class="c0"
  >
    <a
      href="/"
    >
      <img
        alt="Evandro Calado"
        class="c1"
        src="https://avatars.githubusercontent.com/u/110628201?v=4"
      />
      <h1
        class="c2"
      >
        Evandro Calado
      </h1>
    </a>
  </header>
</div>
`;
